tests/chainer_tests/training_tests/triggers_tests/test_interval_trigger.py

```diff
@@ -30,51 +33,81 @@ def epoch_detail(self):
         return self.iteration / self.iters_per_epoch
 
     @property
+    def previous_epoch_detail(self):
+        if self.iteration == 0:
+            return None
+        return (self.iteration - 1) / self.iters_per_epoch
+
+    @property
     def is_new_epoch(self):
         return 0 <= self.iteration % self.iters_per_epoch < 1
 
-
-def _test_trigger(self, updater, trigger, expecteds):
-    trainer = training.Trainer(updater)
-    for expected in expecteds:
-        updater.update()
-        self.assertEqual(trigger(trainer), expected)
-
-
-class TestIterationIntervalTrigger(unittest.TestCase):
-
-    def test_iteration_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=5)
-        trigger = training.trigger.IntervalTrigger(2, 'iteration')
-        expected = [False, True, False, True, False, True, False]
-        _test_trigger(self, updater, trigger, expected)
-
-
-class TestEpochIntervalTrigger(unittest.TestCase):
-
-    def test_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=5)
-        trigger = training.trigger.IntervalTrigger(1, 'epoch')
-        expected = [False, False, False, False, True, False, False]
-        _test_trigger(self, updater, trigger, expected)
-
-
-class TestFractionalEpochIntervalTrigger(unittest.TestCase):
-
-    def test_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=2)
-        trigger = training.trigger.IntervalTrigger(1.5, 'epoch')
-        expected = [False, False, True, False, False, True, False]
-        _test_trigger(self, updater, trigger, expected)
-
-
-class TestUnalignedEpochIntervalTrigger(unittest.TestCase):
-
-    def test_unaligned_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=2.5)
-        trigger = training.trigger.IntervalTrigger(1, 'epoch')
-        expected = [False, False, True, False, True, False, False]
-        _test_trigger(self, updater, trigger, expected)
+    def serialize(self, serializer):
+        self.iteration = serializer('iteration', self.iteration)
+
+
+@testing.parameterize(*testing.product_dict([
+    # iteration
+    {
+        'iters_per_epoch': 5, 'interval': (2, 'iteration'), 'resume': 4,
+        'expected': [False, True, False, True, False, True, False]},
+    # basic epoch
+    {
+        'iters_per_epoch': 1, 'interval': (3, 'epoch'), 'resume': 4,
+        'expected': [False, False, True, False, False, True, False]},
+    # fractional epoch
+    {
+        'iters_per_epoch': 2, 'interval': (1.5, 'epoch'), 'resume': 4,
+        'expected': [False, False, True, False, False, True, False]},
+    # unaligned epoch
+    {
+        'iters_per_epoch': 2.5, 'interval': (1, 'epoch'), 'resume': 3,
+        'expected': [False, False, True, False, True, False, False]},
+    # tiny epoch
+    {
+        'iters_per_epoch': 0.5, 'interval': (1, 'epoch'), 'resume': 4,
+        'expected': [True, True, True, True, True, True, True]},
+]))
```

Comment:
Thank you. I am planning to update the tests for `ManualScheduleTrigger` using this.